src/modbus/ModbusBridge.cpp

```diff
@@ -351,9 +352,39 @@ void ModbusBridge::stop()
 
 void ModbusBridge::run()
 {
+    m_timeoutIterator = 0;
+    m_shouldReconnect = false;
+    m_lastReconnectTime = static_cast<unsigned long long>(
+      std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count());
+
     while (m_readerShouldRun)
     {
-        readAndReportModbusRegistersValues();
+        if (!m_shouldReconnect)
+        {
+            m_timeoutIterator = 0;
+            readAndReportModbusRegistersValues();
+        }
+        else if (m_lastReconnectTime + m_timeoutDurations[m_timeoutIterator] <
+                 static_cast<unsigned long long>(
+                   std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch())
```

Comment:
You can get current time once at the beginning of the loop and use it both here and below